internal/provider/resource_gitlab_group.go

```diff
@@ -208,7 +208,9 @@ func resourceGitlabGroupCreate(ctx context.Context, d *schema.ResourceData, meta
 		options.ParentID = gitlab.Int(v.(int))
 	}
 
-	if v, ok := d.GetOk("default_branch_protection"); ok {
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("default_branch_protection"); ok {
```

Comment:
Not a critique but I struggle with how we should handle default values in the provider. If we were adding this as a new resource, would we use `d.Get` here, in order to allow a zero-value, and set a default value for `default_branch_protection` in the schema? Or use `d.GetOkExists`? Regardless I agree with your handling here for compatibility reasons.

Reply:
Right, `GetOk()` doesn't work on zero-values. I was more just seeing if you would help me touch grass here. Since `GetOkExists()` is apparently discouraged, if we were starting over on `gitlab_group` from scratch, would you still use it? Or would you use `Get()` and set an opinionated default in the provider?

Reply:
hmmm 🤔 

> Or would you use Get() and set an opinionated default in the provider?

So, just to make sure I understand correctly, this will only work if this "opinionated default in the provider" is set to something we'd always know it's the "optional default" and the user would never set this in the config themselves, right?
And if it's set to that specific default, we'd not send the field.
The problem is that `Get()` will always return something and we'd never know if we have to include it in the request or not.

I think that even when writing the resource from scratch I'd still use `GetOkExists()` as long as it's working properly, even if discouraged ... I don't see another way to implement this functionality at the moment ..

Reply:
I'm saying we could use `Get()` and always send the field. And since the API docs say that the default is `2`, we could make that our default also.

Anyway this is hypothetical so I'll resolve. 😄

Reply:
oh, shoot you are right 🤦  I somehow had in mind that it would use some default set on instance-level which we wouldn't know in the provider ... that seems to be wrong though. Must have confused it with another resource 🤔 

In that case, yes, we could use `Get()` ..

Reply:
oh, that's where my confusion originated: 

![image](https://user-images.githubusercontent.com/1008252/153558424-cc0f339f-b1cd-4f5a-8519-bb07a0ead253.png)

I'm not sure if `2` really is the default or just what the default of a pristine GitLab instance is 🤔

Reply:
I guess I'm not sure if we care if the GitLab instance is configured with a different default. I'll probably research some other providers when I get a chance.

Reply:
hmmm ... I think here it's not too crucial, but I think we (or someone else) at some point will care 🙃